lib/widgets/message_list.dart

```diff
@@ -510,6 +514,68 @@ class MarkAsReadWidget extends StatelessWidget {
   }
 }
 
+class _TypingStatusState extends State<TypingStatusWidget> with PerAccountStoreAwareStateMixin<TypingStatusWidget> {
+  TypingStatus? model;
+
+  @override
+  void onNewStore() {
+    model?.removeListener(_modelChanged);
+    model = PerAccountStoreWidget.of(context).typingStatus
+      ..addListener(_modelChanged);
+  }
+
+  @override
+  void dispose() {
+    model?.removeListener(_modelChanged);
+    super.dispose();
+  }
+
+  void _modelChanged() {
+    setState(() {
+      // The actual state lives in [model].
+      // This method was called because that just changed.
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final store = PerAccountStoreWidget.of(context);
+    final narrow = widget.narrow;
+    const placeholder = SizedBox(height: 8);
+    if (narrow is! SendableNarrow) return placeholder;
+
+    final typistNames = model!.typistIdsInNarrow(narrow)
+      .where((id) => id != store.selfUserId)
+      .map((id) => store.users[id]!.fullName)
+      .toList();
+    if (typistNames.isEmpty) return placeholder;
+
+    final localization = ZulipLocalizations.of(context);
+    final String text = switch (typistNames.length) {
+      1 => localization.onePersonTyping(typistNames[0]),
+      2 => localization.twoPeopleTyping(typistNames[0], typistNames[1]),
+      _ => localization.manyPeopleTyping,
+    };
+
+    return Padding(
+      padding: const EdgeInsets.only(left: 16, top: 2),
```

Comment:
Since left and right aren't equal, this looks like a place to use `EdgeInsetsDirectional`.